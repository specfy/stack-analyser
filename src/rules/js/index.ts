import './alpinejs.js';
import './angular.js';
import './bootstrap.js';
import './emberjs.js';
import './esbuild.js';
import './eslint.js';
import './express.js';
import './fastify.js';
import './koa.js';
import './meteorjs.js';
import './mithriljs.js';
import './nextjs.js';
import './nuxtjs.js';
import './preactjs.js';
import './prettier.js';
import './prisma.js';
import './puppeteer.js';
import './react.js';
import './reactEmail.js';
import './rollup.js';
import './sequelize.js';
import './socketio.js';
import './solidjs.js';
import './storybook.js';
import './strapi.js';
import './stylelint.js';
import './sveltejs.js';
import './tailwind.js';
import './typescript.js';
import './vite.js';
import './vue.js';
import './webpack.js';
