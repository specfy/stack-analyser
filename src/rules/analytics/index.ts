import './amplitude.js';
import './googleanalytics.js';
import './logsnag.js';
import './matomo.js';
import './mixpanel.js';
import './plausible.js';
import './tinybird.js';
import './vercel.analytics.js';
