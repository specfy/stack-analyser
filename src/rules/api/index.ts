import './gcp.aiplatform.js';
import './gcp.dialogflow.js';
import './gcp.kms.js';
import './gcp.language.js';
import './gcp.maps.js';
import './gcp.speech.js';
import './gcp.translate.js';
import './gcp.vision.js';
import './sitecore.xmcloud.js';
import './vercel.ai.js';
