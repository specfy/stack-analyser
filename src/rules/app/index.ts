import './caddy.js';
import './discourse.js';
import './grafana.js';
import './httpd.js';
import './kibana.js';
import './nginx.js';
import './zookeeper.js';
