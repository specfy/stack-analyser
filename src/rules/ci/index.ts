import './appveyor.js';
import './azure.ci.js';
import './browserstack.js';
import './circleci.js';
import './cirrusci.js';
import './codesandboxci.js';
import './cypressci.js';
import './dependabot.js';
import './github.actions.js';
import './gitlab.ci.js';
import './jenkins.js';
import './relativeci.js';
import './renovate.js';
import './styleci.js';
import './teamcity.js';
import './travisci.js';
