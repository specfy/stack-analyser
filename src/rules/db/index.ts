import './aws.dynamodb.js';
import './cassandra.js';
import './cockroachdb.js';
import './couchbase.js';
import './elasticsearch.js';
import './gcp.bigquery.js';
import './gcp.sql.js';
import './influxdb.js';
import './mariadb.js';
import './memcached.js';
import './mongo.js';
import './mysql.js';
import './neo4j.js';
import './percona.js';
import './postgres.js';
import './redis.js';
import './snowflake.js';
import './vault.js';
