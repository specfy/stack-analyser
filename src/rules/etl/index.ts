import './airbyte.js';
import './apache_airflow.js';
import './apache_flink.js';
import './apache_storm.js';
import './databricks.js';
import './dataiku.js';
import './gcp.dataproc.js';
import './gcp.dataflow.js';
