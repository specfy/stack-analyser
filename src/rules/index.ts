import './analytics/index.js';
import './api/index.js';
import './app/index.js';
import './ci/index.js';
import './db/index.js';
import './etl/index.js';
import './hosting/index.js';
import './js/index.js';
import './messaging/index.js';
import './network/index.js';
import './php/index.js';
import './saas/index.js';
import './spec/index.js';
import './storage/index.js';
import './tool/index.js';
