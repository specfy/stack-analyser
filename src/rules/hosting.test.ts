import { describe, expect, it } from 'vitest';

import { analyser } from '../analyser/index.js';
import { listIndexed } from '../common/techs.js';
import { FakeProvider } from '../provider/fake.js';
import { rawList } from '../rules.js';
import './index.js';

describe('hosting', () => {
  it('should match everything with files', async () => {
    const paths: string[] = [];
    for (const item of rawList) {
      if (
        item.type !== 'file' ||
        listIndexed[item.ref.tech].type !== 'hosting' ||
        !item.ref.files
      ) {
        continue;
      }

      paths.push('example' in item.ref ? item.ref.example : item.ref.files[0]);
    }

    const res = await analyser({
      provider: new FakeProvider({
        paths: {
          '/': paths,
        },
        files: {},
      }),
    });
    expect(res.toJson('').techs).toMatchSnapshot();
  });
});
