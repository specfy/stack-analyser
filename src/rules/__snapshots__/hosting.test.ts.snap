// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`hosting > should match everything with files 1`] = `
[
  "expodev",
  "flyio",
  "github.pages",
  "heroku",
  "netlify",
  "platformsh",
  "render",
  "vercel",
]
`;
